implementation/src/main/java/io/smallrye/mutiny/operators/multi/overflow/MultiOnOverflowBufferOp.java

```diff
@@ -70,7 +71,7 @@ public void onSubscribe(Subscription subscription) {
 
         @Override
         public void onItem(T t) {
-            if ((!unbounded && Queues.isOverflowing(queue, bufferSize)) || !queue.offer(t)) {
+            if ((!unbounded && strictBoundCounter.getAndIncrement() >= bufferSize) || !queue.offer(t)) {
```

Comment:
same as before, I suggest to expose a specific API, to help writing a more optimized version for the common use case (I can help there, to provide something specific)